TYPO3.Neos/Classes/TYPO3/Neos/EventLog/Domain/Repository/EventRepository.php

```diff
@@ -59,9 +59,7 @@ public function removeAll()
         $classMetaData = $this->entityManager->getClassMetadata($this->getEntityClassName());
         $connection = $this->entityManager->getConnection();
         $databasePlatform = $connection->getDatabasePlatform();
-        $connection->query('SET FOREIGN_KEY_CHECKS=0');
         $truncateTableQuery = $databasePlatform->getTruncateTableSql($classMetaData->getTableName());
         $connection->executeUpdate($truncateTableQuery);
-        $connection->query('SET FOREIGN_KEY_CHECKS=1');
```

Comment:
AFAIR it's when you have nested events it starts to create trouble, so it's likely the tests are and your manual testing won't reveal it.. don't think it's about other tables, just nested references on itself